Ct 4208 anonymise cases

willmcb opened the pull request:
## Description
PR is first of two to do with anonymisation for Branston cases. This PR includes a service that anonymises cases and its associated spec.

As such there is no user testing to be done just a code review.

A second PR will be raised to orchestrate individually feeding cases into this service to process their anonymisation.:w
 

## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed


### Screenshots
n/a

### Related JIRA tickets
[CT-4208](https://dsdmoj.atlassian.net/browse/CT-4208)

### Deployment
<!-- Notes about database migrations, new runtime dependencies, mitigating downtime, feature flags, etc -->

### Manual testing instructions
- run the spec, and a code review please :)

Commit:
Basic kase values anonymised

Commit:
WIP

Commit:
Anon case notes

Commit:
add ability to destroy papertrail versions

Commit:
fix typos

Commit:
Changes cases RetentionSchedule state on anonymisation

There was an issue with the names of the states 'destroyed' was
conflicting with rails methods so the record was unable to save.

RetentionSchedule states now use 'anonymise' as the base word though what the
user sees is the same.

Commit:
Add guard clauses / exceptions

Commit:
Fix filter issue + rubocop

Commit:
Fix specs

zheileman reviewed (approved):
👍 LGTM
Just a few suggestions/comments!

Commit:
Update email values

Commit:
Remove papertrail logs from anonymisation updates

Also uses update_attribute to avoid validations as they are irrelivant
at this point.

Commit:
remove unnecessary if statements

Commit:
WIP

Commit:
Add test for third party cases

Also chances some anon vals to match those in the requirements

Also respects requirement to not add anon vals in places where original
case value was nil.
